### PR TITLE
Update GLWrapper to check for IE

### PR DESCRIPTION
**Pull request by katydecorah**
This PR:

- Update mapbox-gl-supported to the latest.
- Reworks the message for GLWrapper to use `notSupportedReason` provided by mapbox-gl-supported:
  - If IE 11, show our message about IE 11 and Mapbox GL JS support.
  - If no WebGL support, show original GLWrapper message.
  - else let the user know the reason supplied by `notSupportedReason`. (There can be [many reasons](https://github.com/mapbox/mapbox-gl-supported/blob/214c1ed12dba316a6434469384010948354a8c69/index.js#L18-L30) why Mapbox GL is unsupported that are unrelated to WebGL or IE!)
- Adds an optional prop to GlWrapper: `reason`. This allows us to create (non-rendering) test cases for IE and no WebGL support to generate snapshot tests. This will be helpful in case future mapbox-gl-supported versions change the handling of these two reasons.

## How to test

Because Jest doesn't have WebWorker support, it makes it difficult to generate separate tests for IE 11 support, no WebGl support, and any other messages. (You'll noticed that the snapshot tests updated to state the reason for GL unsupported is due to `Mapbox GL is unsupported due to insufficient worker support.`)

You can also manually test the GlWrapper:

1. Run test-cases app and open /GlWrapper (npm start)
2. Open page in IE 11 via BrowserStack, it should look like:
image

One way to check for WebGL support is to disable it in Firefox (temporarily):

1. In Firefox go to `about:config`
2. Search for `webgl.disable` and set it to `true.
3. Open page in Firefox, it should look like:
image

## QA checklist

Open the test cases app locally on:

- [x] Chrome, no errors logged to console.
- [x] Firefox, no errors logged to console.
- [x] Safari, no errors logged to console.
- [x] Edge, no errors logged to console.
- [x] IE11, no errors logged to console.
- [x] Mobile Safari, no errors logged to console.
- [x] Android Chrome, no errors logged to console.

Fixes #387

### CODE REVIEW

File: src/components/gl-wrapper/__tests__/__snapshots__/gl-wrapper.test.js.snap [hunk 1]
```diff
@@ -1,6 +1,142 @@
...
+  </div>
+  <div
+    className="flex-child prose"
+  >
+    <div
+      className="txt-bold txt-m mb6"
+    >
+      Mapbox GL unsupported
+    </div>
+    Mapbox GL is unsupported due to insufficient worker support.
```
Comment (katydecorah) [inline src/components/gl-wrapper/__tests__/__snapshots__/gl-wrapper.test.js.snap:51]: 📝 We expect the "basic" test case app example to return this message as Jest isn't a browser and doesn't have WebWorker support.

File: src/components/gl-wrapper/__tests__/__snapshots__/gl-wrapper.test.js.snap [hunk 2]
```diff
@@ -1,6 +1,142 @@
...
+  </div>
+  <div
+    className="flex-child prose"
+  >
+    <div
+      className="txt-bold txt-m mb6"
+    >
+      Mapbox GL unsupported
+    </div>
+    Starting with v2.0.0, Mapbox GL JS is no longer compatible with any version of Internet Explorer. If you need to support Internet Explorer, consider using the
```
Comment (katydecorah) [inline src/components/gl-wrapper/__tests__/__snapshots__/gl-wrapper.test.js.snap:104]: 📝 This non-rendering test case example sets the reason manually to `insufficient ECMAScript 6 support` so we can track the generated warning message for IE.

File: src/components/gl-wrapper/__tests__/gl-wrapper-test-cases.js [hunk 3]
```diff
@@ -9,4 +10,22 @@ testCases.basic = {
...
+
+noRenderCases.webgl = {
+  description: "reason='insufficient WebGL support'",
+  element: (
+    <GLWrapper reason="insufficient WebGL support">
+      Your browser supports Mapbox GL!
+    </GLWrapper>
+  )
+};
+
```
Comment (katydecorah) [inline src/components/gl-wrapper/__tests__/gl-wrapper-test-cases.js:30]: 📝 This `noRenderCases` generate a snapshot so we can track if there are any changes the next time we update mapbox-gl-supported.

**Review by colleenmcginnis — APPROVED** [review]
🎉

**Pull request merged**